### PR TITLE
GODRIVER-1762 up timeout

### PR DESCRIPTION
**Pull request by iwysiu**
Patch with the test timeout upped to 30 mins [all passed](https://spruce.mongodb.com/version/601ad3c7d6d80a3247c08878/)

### CODE REVIEW

**Comment by divjotarora** [comment]
There's at least one failure in the checks linked to this PR. Should we investigate those before merging this as-is?

**Comment by iwysiu** [comment]
> There's at least one failure in the checks linked to this PR. Should we investigate those before merging this as-is?

From a quick look, that error isn't timeout related, and I've never actually seen it before. I'm looking into it but i think its separate from these changes

**Review by benjirewis — APPROVED** [review]
Seems fine to me; the one error seems related to `WaitQueueTimeoutError` which I updated in a recent commit, but I can't recreate it locally... I'll keep looking into it.

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ PKGS = $(BSON_PKGS) $(EVENT_PKGS) $(MONGO_PKGS) $(UNSTABLE_PKGS) $(TAG_PKG) $(EX
 TEST_PKGS = $(BSON_TEST_PKGS) $(EVENT_TEST_PKGS) $(MONGO_TEST_PKGS) $(UNSTABLE_TEST_PKGS) $(TAG_PKG) $(EXAMPLES_TEST_PKGS)
 ATLAS_URIS = "$(ATLAS_FREE)" "$(ATLAS_REPLSET)" "$(ATLAS_SHARD)" "$(ATLAS_TLS11)" "$(ATLAS_TLS12)" "$(ATLAS_FREE_SRV)" "$(ATLAS_REPLSET_SRV)" "$(ATLAS_SHARD_SRV)" "$(ATLAS_TLS11_SRV)" "$(ATLAS_TLS12_SRV)"
 
-TEST_TIMEOUT = 600
+TEST_TIMEOUT = 1800
```
Comment (kevinAlbs) [inline Makefile:17]: It looks like there are two types of timeouts on evergreen:
1. macOS/Windows timeouts, [example](https://evergreen.mongodb.com/task/mongo_go_driver_tests_nonlegacy_servers__version~3.2_os_ssl_32~osx_go_1_15_test_replicaset_auth_nossl_213fb80b373f70dba4f9f516dc4c718abe41c76b_21_02_02_19_12_49) 
2. TestPool timeouts, [example](https://evergreen.mongodb.com/task/mongo_go_driver_tests_legacy_noauth_nossl__version~3.0_os_ssl_32~ubuntu1604_64_go_1_15_test_sharded_noauth_nossl_213fb80b373f70dba4f9f516dc4c718abe41c76b_21_02_02_19_12_49)

I think increasing the timeout addresses (1). Per offline discussion, these timeouts looked like they started occurring consistently after GODRIVER-1715, which added more time to test runs. So I think the deal is that we're just hitting the timeout sometimes, and every test that started after the 10 minute timeout is marked as a failure. That seems totally fine.

I am still unsure about (2). The [last successful run of (2)](https://evergreen.mongodb.com/task/mongo_go_driver_tests_legacy_noauth_nossl__version~3.0_os_ssl_32~ubuntu1604_64_go_1_15_test_sharded_noauth_nossl_1ac4536988c40caeab77b417f2060f10195fa489_21_02_02_18_20_57) ran within 3min.

Did you see (2) in any patch builds? It seems more rare. But I suspect we will see (2) again. But (2) may be best investigated as part of a separate ticket. If most things failing on evergreen are (1), then increasing the timeout to address that seems good.

Can you note in GODRIVER-1762 what we think the cause is? That way future readers will have more context around the value chosen.

**Review by kevinAlbs — COMMENTED** [review]
Yay for green evergreen! I have a question about the root cause. In particular, I am surprised failures for TestPool have not reappeared. But we can do those in another ticket. It seems less urgent if those are rarer.

File: Makefile (hunk 1 above)
Comment (iwysiu) [reply Makefile:17]: Comment added. I didn't see (2) in the patch builds, but i also suspect that it will reoccur in the future.

**Review by kevinAlbs — APPROVED** [review]
Thank you for looking into this!

**Review by divjotarora — APPROVED** [review]

**Pull request merged**